File: 560. Subarray Sum Equals K/Solution.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int ans=0;
        int sum=0;
        unordered_map<int,int> mp;
        mp[0]=1;//assumed that there always exist a empty prefix with zero sum
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            ans+=mp[sum-k];
            mp[sum]++;
        }
        return ans;
    }
};

File: 560. Subarray Sum Equals K/Solution.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mp = {0: 1}
        sum = 0
        ans = 0
        
        for i in nums:
            sum += i
            ans += mp.get(sum-k, 0)
            mp[sum] = 1 + mp.get(sum, 0)

        return ans



